src/validate/methods/secretKey.ts

```diff
@@ -0,0 +1,15 @@
+import { StrKey } from "@stellar/stellar-sdk";
+
+export const secretKey = (value: string) => {
+  if (value.startsWith("S")) {
+    if (!StrKey.isValidEd25519SecretSeed(value)) {
+      return "Invalid secret key.";
+    }
+  } else {
+    if (!value.match(/^[0-9a-f]{2,128}$/gi) || value.length % 2 == 1) {
+      return `Invalid hex value. Please provide up to 64 bytes in hexadecimal format.`;
+    }
+  }
+
+  return false;
```

Comment:
we are using `return false` for `validate` methods because we're using for `error = validate.secretKey` but when I just look at the validate method codes, I expect the validated return value to be `true` instead of `false`

Reply:
We are returning a string (error message) if it is not valid, so it made sense to return `false` or an empty string when it is valid. I chose to use `false` to make it easier to check response type. I agree that it is a bit confusing when looking at the method. Should we rename it to something other than `validate`?

Reply:
how about the following?
```
validators/
  └── methods/
      ├── getAmountValidationError.ts
      ├── getAssetValidationError.ts
      ├── getPositiveIntegerValidationError.ts
```